.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,61 @@
+name: Publish to DockerHub
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  release:
+    types: [created]
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: ConservationMetrics/superset-deployment
+      - name: Extract Superset version from Dockerfile
+        id: version
+        run: |
+          SS_VERSION=$(grep "FROM apache/superset:" Dockerfile | cut -d':' -f2)
+          echo "ss_version=${SS_VERSION}" >> $GITHUB_OUTPUT
+      - name: Generate timestamp
+        id: timestamp
+        run: |
+          BUILD=$(date +"%Y%m%d-%H%M")
+          echo "build=${BUILD}" >> $GITHUB_OUTPUT
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      # Push for main branch or same-repo PRs (security measure for external PRs)
+      # PR builds: pr-123-4.1.2_20250115-1234 (for testing)
```

Comment:
I am thinking that this PR build never happens because the entire workflow only runs `on.push.branches.'main'` (lines 5-6)